apps/sms/views/shared/js/activity_picker.js

```diff
@@ -3,87 +3,119 @@
 (function(exports) {
 'use strict';
 
-function handleActivity(activity, onsuccess, onerror) {
-  // Note: the MozActivity is intentionally constructed in the caller and
-  // passed down to this function in order to make it possible to pass an
-  // inline option to its constructor in the caller for the sake of static
-  // analysis tools.  Please do not change that!
+  /**
+   * ActivityPicker
+   *
+   * A collection of some MozActivity methods.
+   */
+  var ActivityPicker = {
 
-  if (typeof onsuccess === 'function') {
-    activity.onsuccess = onsuccess;
-  }
+    /**
+     * When user wants to pass a phone call.
+     * @param  {Number} contact number where to dial.
+     * @returns {Promise} DOMRequest promise is always successfully resolved.
+     */
+    dial(number) {
+      return Promise.resolve(new MozActivity({
+        name: 'dial',
+        data: {
+          type: 'webtelephony/number',
+          number: number
+        }
+      }));
+    },
 
-  activity.onerror = typeof onerror === 'function' ? onerror : function(error) {
-    console.warn('Unhandled error spawning activity; ' + error.message);
-  };
-}
+    /**
+     * When user wants to send a new Email. The e-mail app can parse mailto 
+     * URI strings provided as either a "url" or "URI" property.
+     * @param  {String} Email id of receiver.
+     * @returns {Promise} DOMRequest promise is always successfully resolved.
+     */
+    email(email) {
+      return Promise.resolve(new MozActivity({
+        name: 'new',
+        data: {
+          type: 'mail',
+          URI: 'mailto:' + email
+        }
+      }));
+    },
+
+    /**
+     * When user wants to open a URL.
+     * @param  {String} URL to be visited.
+     * @returns {Promise} DOMRequest promise is always successfully resolved
+     */
+    url(url) {
+      return Promise.resolve(new MozActivity({
+        name: 'view',
+        data: {
+          type: 'url',
+          url: url
+        }
+      }));
+    },
+
+    /**
+     * When user wants to create a new contact entry.
+     * @param  {Object} Contact object.
```

Comment:
It might be a little bit misleading if we commented that it's a contact object. contact object might be a real contact object returned from MozContacts API. It's only the object that contains contact information (phone or email), and it's different from the object we set for  viewContact.